### ISSUE TITLE
request help: why the location in response header that returned by APISIX has the 9443 port?

### ISSUE THREAD

**Issue by shuoshadow** [apache/apisix#6954]
### Issue description
apisix前使用slb 443端口转发到apisix的9443端口
apisix 代理https请求后 携带了9443端口导致无法重定向
![image](https://user-images.githubusercontent.com/10016360/120430719-01fac380-c3aa-11eb-98a6-66440e9f51b7.png)

是因为header的原因吗？我配置了apisixroute的proxy-rewrite插件也不行
```yaml
    - name: proxy-rewrite
      enable: true
      config:
        headers:
          "Host": $host
```
### Environment

* your apisix-ingress-controller version (output of `apisix-ingress-controller version --long`);
* your Kubernetes cluster version (output of `kubectl version`);
* if you run apisix-ingress-controller in Bare-metal environment, also show your OS version (`uname -a`).

**Comment by Horus-K** [comment]
@shuoshadow Temporary workaround: Change the apisix port to 80 443

**Comment by tokers** [comment]
Or should apache apisix handle the port in redirect url by using the X-Forwarded-Port?

**Comment by tzssangglass** [comment]
> Or should apache apisix handle the port in redirect url by using the X-Forwarded-Port?

+1.

ref: #6945

**Comment by snksos** [comment]
https://apisix.apache.org/docs/apisix/plugins/redirect#example-usage

**Comment by vvavepacket** [comment]
Was this fixed on recent releases? Whats the recommended way going forward for this issue?

**Comment by tokers** [comment]
> Was this fixed on recent releases? Whats the recommended way going forward for this issue?

What's your version? Recent releases will check the `X-Forwarded-Port` header, so the redirect will be correct.

**Comment by fbossiere** [comment]
I am using the last available version of the apisix helm chart with default option.
When i try to access the keycloak admin console, APISIX redirects me as follows:
`https://keycloak.mydomain.com:9443/admin/`

Can you please give us the way to fix this please ?

**Comment by tokers** [comment]
@fbossiere Without a context, I didn't know if this is an issue or a normal scene.

**Comment by fbossiere** [comment]
The context is very simple.

1- You deploy apisix ingress controller on any 1.2x k8s cluster using the
official documentation with default values yaml, then
2- you helm install keycloak with default values.yaml, together with
3- a standard apisix route + certificate

Then you notice that this weird redirection occurs when you try to access
the admin console of keycloak.

Something is clearly going wrong with the X-forwarded-port but nothing in
apisix-ingress-controller documentation tells us how to fix.

This problem doesn't occur with nginx ingress controller, nor traefik. It
basically has been preventing our company to use apisix until now.

**Comment by tokers** [comment]
a quick fix for this is setting the HTTPS port to `443` via the field: https://github.com/apache/apisix-helm-chart/blob/master/charts/apisix/values.yaml#L192.

**Comment by tokers** [comment]
@fbossiere If this is OK for you, would you like to submit a PR to mention it in the docs?

**Comment by lkad** [comment]
"Has anyone fixed this issue? I am experiencing the same problem."

**Comment by lkad** [comment]
> a quick fix for this is setting the HTTPS port to `443` via the field: https://github.com/apache/apisix-helm-chart/blob/master/charts/apisix/values.yaml#L192.

This configuration has the load balancer port set to 443, but if the port inside the container is 9443, it will still redirect to 9443.

**Comment by vinson-zhang** [comment]
+1

**Comment by Eifoen** [comment]
+1
